File: BasicAppSample/app/src/javaVersion/java/com/example/basicappsample/helpers/KeyboardHelpable.java
package com.example.basicappsample.helpers;

import android.app.Activity;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public interface KeyboardHelpable {
    default void hideKeyboard(Activity activity) {
        final View currentFocus = activity.getCurrentFocus();
        if (currentFocus != null) {
            final InputMethodManager inputMethodManager =
                    (InputMethodManager) activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
            inputMethodManager.hideSoftInputFromWindow(currentFocus.getWindowToken(), 0);
        }
    }
}

File: BasicAppSample/app/src/kotlinVersion/java/com/example/basicappsample/helpers/KeyboardHelpable.kt
package com.example.basicappsample.helpers

import android.app.Activity
import android.view.inputmethod.InputMethodManager

interface KeyboardHelpable {
    fun Activity.hideKeyboard() {
        this.currentFocus?.let {
            val inputMethodManager =
                this.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
            inputMethodManager.hideSoftInputFromWindow(it.windowToken, 0)
        }
    }
}